safehttp/response_writer_test.go

```diff
@@ -125,3 +125,103 @@ func TestResponseWriterWriteTwicePanic(t *testing.T) {
 		})
 	}
 }
+
+func TestResponseWriterStatusCode(t *testing.T) {
+	tests := []struct {
+		name  string
+		want  safehttp.StatusCode
+		write func(w *safehttp.ResponseWriter)
+	}{
+		{
+			name: "Status code not explicitly set",
+			want: safehttp.StatusOK,
+			write: func(w *safehttp.ResponseWriter) {
+				w.Write(safehtml.HTMLEscaped("<h1>Escaped, so not really a heading</h1>"))
+			},
+		},
+		{
+			name: "Custom status code before calling write",
+			want: safehttp.StatusCreated,
+			write: func(w *safehttp.ResponseWriter) {
+				w.Code = safehttp.StatusCreated
+				w.Write(safehtml.HTMLEscaped("<h1>Escaped, so not really a heading</h1>"))
+			},
+		},
+		{
+			name: "Custom status code after calling write",
+			want: safehttp.StatusOK,
+			write: func(w *safehttp.ResponseWriter) {
+				w.Write(safehtml.HTMLEscaped("<h1>Escaped, so not really a heading</h1>"))
+				w.Code = safehttp.StatusCreated
+			},
```

Comment:
Why is this considered good?

Reply:
it's true it shouldn't be allowed and should probably trigger an error